packages/template-retail-react-app/app/hooks/use-dnt-notification.js

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage, useIntl} from 'react-intl'
+import {
+    Button,
+    Modal,
+    ModalContent,
+    ModalBody,
+    ModalCloseButton,
+    useDisclosure,
+    Heading,
+    Stack,
+    Text,
+    Flex
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {HideOnDesktop, HideOnMobile} from '@salesforce/retail-react-app/app/components/responsive'
+import {useDNT} from '@salesforce/commerce-sdk-react'
+import {useLocation} from 'react-router-dom'
+
+export const DntNotification = ({isOpen, onOpen, onClose}) => {
+    const {dntStatus, updateDNT} = useDNT()
+    const {formatMessage} = useIntl()
+    const location = useLocation()
+
+    useEffect(() => {
+        if (dntStatus === undefined) {
+            onOpen()
+        } else {
+            onClose()
+        }
+    }, [location, dntStatus])
+
+    const onCloseNotification = async () => {
+        await updateDNT(null)
+        onClose()
```

Comment:
```suggestion
        updateDNT(null)
        onClose()
```

There are several calls to `updateDNT` that are followed by closing the notification 🤔  Yes, `updateDNT` is an async operation, but I'm not sure if awaiting it will be good for the user experience.

Usually the act of clicking a button should result in immediate feedback. But in this case, it's possible to see a delay, while we wait for `updateDNT` to finish. If it's not crucial to await, I'd prefer to not await and let `updateDNT` run in the background.

Reply:
@alexvuong recommended we await the updateDNT in case another reauthentication happens around the same time, at which point if we don't await this call then that second call could conflict and overwrite the token's DNT and we could lose the update here.